luaui/Widgets/gui_spectator_hud.lua

```diff
@@ -0,0 +1,2288 @@
+
+		local bottom = vsModeMetricsAreaTop - metric.id * vsModeMetricHeight
+
+		local iconLeft = vsModeMetricsAreaLeft + borderPadding + vsModeMetricIconPadding
+		local iconRight = iconLeft + vsModeMetricIconWidth
+		local iconBottom = bottom + borderPadding + vsModeMetricIconPadding
+		local iconTop = iconBottom + vsModeMetricIconHeight
+
+		local iconHCenter = mathfloor((iconRight + iconLeft) / 2)
+		local iconVCenter = mathfloor((iconTop + iconBottom) / 2)
+		local iconText = metric.text
+
+		font:Begin()
+			font:SetTextColor(textColorWhite)
+			font:Print(
+				iconText,
+				iconHCenter,
+				iconVCenter,
+				fontSizeVSBar,
+				'cvo'
+			)
+		font:End()
+
+		local leftKnobLeft = iconRight + borderPadding + vsModeMetricIconPadding * 2
+		local leftKnobBottom = iconBottom
+		local leftKnobRight = leftKnobLeft + vsModeKnobWidth
+		local leftKnobTop = iconTop
+		drawVSModeKnob(
+			leftKnobLeft,
+			leftKnobBottom,
+			leftKnobRight,
+			leftKnobTop,
+			vsModeStats[indexLeft].colorKnobSide,
+			formatResources(statsLeft.total, true)
+		)
+
+		local rightKnobRight = vsModeMetricsAreaRight - borderPadding - vsModeMetricIconPadding * 2
+		local rightKnobBottom = iconBottom
+		local rightKnobLeft = rightKnobRight - vsModeKnobWidth
+		local rightKnobTop = iconTop
+		drawVSModeKnob(
+			rightKnobLeft,
+			rightKnobBottom,
+			rightKnobRight,
+			rightKnobTop,
+			vsModeStats[indexRight].colorKnobSide,
+			formatResources(statsRight.total, true)
+		)
+
+		drawVSBar(
+			leftKnobRight,
+			iconBottom,
+			rightKnobLeft,
+			iconTop,
+			indexLeft,
+			indexRight,
+			metric.id
+		)
+	end
+end
+
+local function mySelector(px, py, sx, sy)
+	-- modified version of WG.FlowUI.Draw.Selector
+
+	local cs = (sy-py)*0.05
+	local edgeWidth = mathmax(1, mathfloor((sy-py) * 0.05))
+
+	-- faint dark outline edge
+	rectRound(px-edgeWidth, py-edgeWidth, sx+edgeWidth, sy+edgeWidth, cs*1.5, 1,1,1,1, { 0,0,0,0.5 })
+	-- body
+	rectRound(px, py, sx, sy, cs, 1,1,1,1, { 0.05, 0.05, 0.05, 0.8 }, { 0.15, 0.15, 0.15, 0.8 })
+
+	-- highlight
+	glBlending(GL.SRC_ALPHA, GL.ONE)
+	-- top
+	rectRound(px, sy-(edgeWidth*3), sx, sy, edgeWidth, 1,1,1,1, { 1,1,1,0 }, { 1,1,1,0.035 })
+	-- bottom
+	rectRound(px, py, sx, py+(edgeWidth*3), edgeWidth, 1,1,1,1, { 1,1,1,0.025 }, { 1,1,1,0  })
+	glBlending(GL.SRC_ALPHA, GL.ONE_MINUS_SRC_ALPHA)
+
+	-- button
+	rectRound(px, py, sx, sy, cs, 1, 1, 1, 1, { 1, 1, 1, 0.06 }, { 1, 1, 1, 0.14 })
+	--WG.FlowUI.Draw.Button(sx-(sy-py), py, sx, sy, 1, 1, 1, 1, 1,1,1,1, nil, { 1, 1, 1, 0.1 }, nil, cs)
+end
+
+local function drawMetricChange()
+	mySelector(
+		headerDimensions.left,
+		metricChangeBottom,
+		headerDimensions.right,
+		headerDimensions.bottom
+	)
+
+	-- TODO: this is not working, find out why
+	local mouseX, mouseY = spGetMouseState()
+	if (mouseX > headerDimensions.left) and
+			(mouseX < headerDimensions.right) and
+			(mouseY > headerDimensions.bottom) and
+			(mouseY < metricChangeBottom) then
```

Comment:
What's not working here?

Mouse position checks like this can be simplified a fair bit by using a construct like `Rect` from gridmenu https://github.com/beyond-all-reason/Beyond-All-Reason/blob/master/luaui/Widgets/gui_gridmenu.lua#L154

This can be used for nice concise checks just going `header:contains(mouseX, mouseY)`. Again not a critical thing, and maybe not worth doing until I abstract that construct somewhere accessible by other widgets, but I haven't really gotten around to it and it may never happen with RMLUI around the corner.